### ISSUE TITLE
[AMDGPU] gfx950 miscompile: #204238 (S_CSELECT for uniform i1 extensions) under `-amdgpu-use-amdgpu-trackers`

### ISSUE THREAD

**Issue by yuyzhang512** [llvm/llvm-project#221911]
On gfx950 (MI350X), **#204238 (`55151885`, "Select S_CSELECT_B32/64 for uniform extensions from i1")** miscompiles a flash-attention forward kernel: ~70% of outputs wrong (max abs err ~1.06 vs 0.01 tol), deterministic, at `-O3` with `-amdgpu-use-amdgpu-trackers`. Correct with the flag off, at `-O0`, or with #204238 reverted. (Triton enables this flag by default for gfx942/gfx950, so it is user-visible.)

**Bisected (build-verified both directions):** first-bad `55151885` (#204238); parent `b74fc87d` good. Reverting #204238 on recent main `b010a18d` also fixes it.

**Reproduce** — `attn_fwd.ll` attached (single `amdgpu_kernel`), same IR, only the flag changes:

```bash
llc -O3 -mtriple=amdgcn-amd-amdhsa -mcpu=gfx950 -amdgpu-use-amdgpu-trackers=1 attn_fwd.ll  # wrong
llc -O3 -mtriple=amdgcn-amd-amdhsa -mcpu=gfx950 -amdgpu-use-amdgpu-trackers=0 attn_fwd.ll  # correct
```

The wrong result is a runtime property (verified on gfx950); the two `.s` differ only in scheduling.

**Localization:** #204238 adds exactly **2** `s_cselect` (the i1-extension patterns) to this kernel; removing just those fixes it. `S_CSELECT` already declares `Uses = [SCC]`, so this is not a missing-`Uses`. 

IR is ~12.6k lines (not reduced — the interestingness test needs gfx950 hardware; happy to share the runtime harness or try `llvm-reduce` with guidance).

[attn_fwd.ll.txt](https://github.com/user-attachments/files/31944131/attn_fwd.ll.txt)

**Comment by yuyzhang512** [comment]
## SCC is NOT clobbered between def and use

This is the part I want to correct in the original report. I assumed a scheduling/SCC-liveness interaction. Checking the final assembly at every occurrence of the new pattern, with a backward scan for the nearest SCC writer:

```
site line 1172
  SCC def : s_and_b64 s[0:1], s[0:1], exec
  between : 2 instr -> v_and_b32_e32, v_writelane_b32
  clobbers: NONE - SCC intact

site line 16171
  SCC def : s_and_b64 s[0:1], s[0:1], exec
  between : 0 instr -> (none, adjacent)
  clobbers: NONE - SCC intact
```

Only VALU instructions separate the def from the use, and those write VCC, not SCC. At the second site they are adjacent. **The `S_CSELECT` reads the correct condition in both cases.** So the new instructions are not themselves miscompiled, and this is not a missing-`Uses`/liveness problem.

**Comment by yuyzhang512** [comment]
close by https://github.com/llvm/llvm-project/issues/225054

**Comment by EugeneZelenko** [comment]
> close by [#225054](https://github.com/llvm/llvm-project/issues/225054)

It's not pull request, so please elaborate.

**Comment by yuyzhang512** [comment]
> > close by [#225054](https://github.com/llvm/llvm-project/issues/225054)
> 
> It's not pull request, so please elaborate.

The reason is that I previously mislocated the root cause of the issue to this PR. The change to `S_CSELECT` only happened to fix the problem because it disturbed the scheduling and register allocation.

Another colleague found that the issue disappears when all SGPR spills are reduced. With the help of AI, he also found that there was an SGPR spill-related stack overwrite. This explanation makes much more sense than the original one.

So I decided to close this PR.
